grails-app/services/com/unifina/service/StreamService.groovy

```diff
@@ -310,9 +310,24 @@ class StreamService {
 		}
 	}
 
+	private Date addThresholdHours(Date date, int hours) {
```

Comment:
To encapsulate the logic for the special value `0` and keep the code DRY with regard to the threshold calculation, how about adding a helper function to `Stream.groovy`:

```
boolean isStale(Date now, Date latestDataTimestamp) {
    if (inactivityThresholdHours == 0) {
        return false
    } else {
	Calendar calendar = Calendar.getInstance()
	calendar.setTime(now)
	calendar.add(Calendar.HOUR_OF_DAY, -inactivityThresholdHours)
	Date threshold = calendar.getTime()

        return latestDataTimestamp.before(threshold)
    }
}
```